zone/waypoints.cpp

```diff
@@ -795,6 +795,10 @@ void Mob::FixZ(int32 z_find_offset /*= 5*/, bool fix_client_z /*= false*/) {
 		return;
 	}
 
+	if (IsBoat()) {
```

Comment:
Since this is running through a significant hot path, I'd like this to reference a mob scoped private member variable boolean that is loaded on spawn and we don't have to run OR's in this check repeatedly

Otherwise I'm good with the overall premise and change of this PR

Reply:
Ok, family here today.  Will make those changes when I get back to it.

Reply:
Ok, is this what you had in mind @Akkadius?  Still compiling.